datameta/api/appsettings.py

```diff
@@ -0,0 +1,174 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+from pyramid.view import view_config
+from pyramid.request import Request
+from pyramid.httpexceptions import HTTPNoContent, HTTPForbidden, HTTPBadRequest
+
+from dataclasses import dataclass
+from . import DataHolderBase
+from typing import List, Dict
+from ..models import ApplicationSettings
+from .. import resource, security, errors
+from ..resource import resource_by_id
+from ..settings import get_setting_value_type
+import datetime
+
+@dataclass
+class AppSettingsResponseElement(DataHolderBase):
+    """Class for Appsettings Request Response from OpenApi"""
+    id:         dict
+    key:        str
+    valueType:  str
+    value:      str
+
+@view_config(
+    route_name="appsettings", 
+    renderer='json', 
+    request_method="GET", 
+    openapi=True
+)
+def get(request: Request) -> List[AppSettingsResponseElement]:
+    """Return all Site Settings"""
+    db = request.dbsession
+
+    # Authenticate the user
+    auth_user = security.revalidate_user(request)
+
+    appsettings = db.query(ApplicationSettings)
+
+    # User has to be site admin to access these settings
+    if not auth_user.site_admin:
+        raise HTTPForbidden()
+
+    settings = []
+
+    for setting in appsettings:
+        value, value_type = get_setting_value_type(setting)
+
+        settings.append(
+            AppSettingsResponseElement(
+                id                    =  resource.get_identifier(setting),
+                key                   =  setting.key,
+                valueType             =  value_type,
+                value                 =  value
+            )
+        )
+
+    return settings
+
+@view_config(
+    route_name="appsettings_id",
+    renderer='json',
+    request_method="PUT",
+    openapi=True
+)
+def put(request:Request):
+    """Change a metadataset"""
+    auth_user = security.revalidate_user(request)
+    db = request.dbsession
+    settings_id = request.matchdict["id"]
+
+    # Only site admins can change site settings
+    if not auth_user.site_admin:
+         raise HTTPForbidden()
+
+    target_setting = resource_by_id(db, ApplicationSettings, settings_id)
+
+    key = request.openapi_validated.body["key"]
+    temp, value_type = get_setting_value_type(target_setting)
+    value = request.openapi_validated.body["value"]
+
+    if value_type == 'int':
+        try:
+            value_int = int(value)
+        except ValueError:
+            err = {
+                "exception": "ValidationError",
+            }   
+            response_body = []
+            err["message"] = "You have to provide an integer."
+            response_body.append(err)
+            return HTTPBadRequest(json=response_body)
```

Comment:
We have a helper function for this.

Also make sure you always `raise` the errors, so that potential database changes are discarded.
```suggestion
            raise errors.get_validation_error(["You have to provide an integer."])
```